### ISSUE TITLE
Unable to provision clusters with ec2 node driver when custom endpoints are required

### ISSUE THREAD

**Issue by durellw** [rancher/rancher#33344]
Please provide a method to supply custom endpoints to the ec2 node driver. This is useful for environments with specific compliance requirements, such as using AWS FIPS 140-2 endpoints, connecting to AWS Snowball, SC2S, or C2S environments, or local testing.

**Comment by nickgerace** [comment]
Are you referring to adding more than the regional endpoints? 
Docs: https://docs.aws.amazon.com/general/latest/gr/rande.html

**Comment by durellw** [comment]
Yes, more than the static list of regional endpoints. For example, if a customer wanted to use FIPS endpoints [1], there is no provision that I am aware of that would allow that.

1. https://aws.amazon.com/compliance/fips/

**Comment by nickgerace** [comment]
I've marked this as a release candidate. Thanks for the response! Apologies I missed it initially.

**Comment by kinarashah** [comment]
We need to investigate further to understand the impact of changes and testing to come up with accurate LOE.

**Comment by snasovich** [comment]
Closing this issue as stale
